backoffice/workflows/tests/test_views.py

```diff
@@ -266,3 +273,57 @@ def test_reject_author(self, mock_post):
         )
 
         self.assertEqual(response.status_code, 200)
+
+    def test_restart_full_dagrun(self):
+
+        self.api_client.force_authenticate(user=self.curator)
+        url = reverse(
+            "api:workflows-authors-restart",
+            kwargs={"pk": self.workflow.id},
+        )
+        with patch_requests() as (mock_post, mock_get, mock_delete):
+            response = self.api_client.post(url)
+
+            self.assertEqual(response.status_code, 200)
+
+    def test_restart_a_task(self):
+
+        self.api_client.force_authenticate(user=self.curator)
+        url = reverse(
+            "api:workflows-authors-restart",
+            kwargs={"pk": self.workflow.id},
+        )
+        with patch_requests() as (mock_post, mock_get, mock_delete):
+            response = self.api_client.post(url, json={"task_ids": ["set_workflow_status_to_running"]})
+            self.assertEqual(response.status_code, 200)
+
+    def test_restart_with_params(self):
+
+        self.api_client.force_authenticate(user=self.curator)
+        url = reverse(
+            "api:workflows-authors-restart",
+            kwargs={"pk": self.workflow.id},
+        )
+        with patch_requests() as (mock_post, mock_get, mock_delete):
+            response = self.api_client.post(url, json={"params": {"workflow_id": self.workflow.id}})
+            self.assertEqual(response.status_code, 200)
+
+
+@contextmanager
+def patch_requests():
+    with patch("requests.post") as mock_post, patch("requests.get") as mock_get, patch(
+        "requests.delete"
+    ) as mock_delete:
+
+        # Configure the mock for requests.post
+        mock_post.return_value.status_code = 200
+        mock_post.return_value.json.return_value = {"key": "value"}
+
+        # Configure the mock for requests.get
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.json.return_value = {"data": "some_data"}
+
+        # Configure the mock for requests.delete
+        mock_delete.return_value.status_code = 204
+
+        yield mock_post, mock_get, mock_delete
```

Comment:
we can use [pytest-vcr](https://pytest-vcr.readthedocs.io/en/latest/) for that

Reply:
Example: https://github.com/SCOAP3/scoap3/blob/27dfb1ed1c3c08dd9339c8ec8fd108ca431e1845/scoap3/articles/tests/test_article_compliance.py#L24